From 8fd65e180f897fd612ce3bd8079a3f6d14c128bd Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:20 -0400
Subject: [fs] ext4: don't leave i_crtime.tv_sec uninitialized

Message-id: <1414058125-4183-107-git-send-email-lczerner@redhat.com>
Patchwork-id: 98202
O-Subject: [RHEL 7.1 PATCH v2 106/171] ext4: don't leave i_crtime.tv_sec uninitialized
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 19ea80603715d473600cd993b9987bc97d042e02

 If the i_crtime field is not present in the inode, don't leave the
 field uninitialized.

 Fixes: ef7f38359 ("ext4: Add nanosecond timestamps")
 Reported-by: Vegard Nossum <vegard.nossum@oracle.com>
 Tested-by: Vegard Nossum <vegard.nossum@oracle.com>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Cc: stable@vger.kernel.org
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/ext4.h b/fs/ext4/ext4.h
index 27caad1..5ddb38d 100644
--- a/fs/ext4/ext4.h
+++ b/fs/ext4/ext4.h
@@ -774,6 +774,8 @@ do {                \
  if (EXT4_FITS_IN_INODE(raw_inode, einode, xtime))         \
   (einode)->xtime.tv_sec =            \
    (signed)le32_to_cpu((raw_inode)->xtime);        \
+ else               \
+  (einode)->xtime.tv_sec = 0;           \
  if (EXT4_FITS_IN_INODE(raw_inode, einode, xtime ## _extra))        \
   ext4_decode_extra_time(&(einode)->xtime,         \
            raw_inode->xtime ## _extra);        \
-- 
1.7.1